---
title: "201-ancestral-again"
author: "Mac Campbell"
date: "5/28/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

## Ancestral sequence reconstruction

Let's put these all into outputs/201. First, compute some EVEs that meet our criterion of >= 4 seqs.

```{r}
library(tidyverse)
library(ggtree)
data<-read_tsv("data/epv-side-data.tsv") %>% filter(locus_name !="NK") %>% group_by(locus_name) %>%
  summarize(Count=n()) %>% arrange(-Count) %>% filter(Count >= 4)
data
data %>% arrange(locus_name)

#Returning on 06042020 to create a df that we can attach to trees.

df<-read_tsv("data/epv-side-data.tsv")

#Need something that matches the tip labels. I'll do it in baseR
df$id<-gsub("-","_",df$id)
df$id<-paste0(df$id,"_")

#Ok, I prefer dplyr verbs
df<-df %>% select(-INDEX) %>% mutate(TipLabel=paste0(id,extract_start))
```

Obtain alignments.

```{r}
path<-data %>% mutate(Path=paste0("/Users/mac/github/AAV-GLUE/alignments/eve/tip/",locus_name,".aln.fna")) %>% select(Path)
write_tsv(path, col_names = FALSE, "/Users/mac/github/eve-trees/outputs/201/paths.txt")
```

Only obtained 12 alignments.
There are some differences in naming for loci and *.fna files.   
felid-pinniped vs carnivore
camelid vs camelidae

There are also two entries in the .tsv consulted:
dependo.20-laurasiatheria	9			
dependo.22-laurasiatheria	42			

```{sh, eval=FALSE}
cp /Users/mac/github/AAV-GLUE/alignments/eve/tip/*.fna .
```

In `/Users/mac/github/eve-trees/outputs/201`     
```{sh, eval=FALSE}
#cat paths.txt | while read line; do cp $line .; done;
for f in *.fna; do seqConverter.pl -d$f -ope; done;
for f in *.phylip; do grep -v _con_ $f > `basename $f .phylip`.phy; done;
```

Keep an eye out for a consensus sequence so that the phylips may have too many taxa specified. Yeah.     

### dependo.1
For dependo.1 we can root by the hippo, Hippopotamus_amphibius_Dependo_166_ here.
```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Hippopotamus_amphibius_Dependo_166_ -n dependo.1 -s dependo.1-whippomorpha.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.1 -s dependo.1-whippomorpha.aln.phy -n dependo.1.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.1.ancestral | perl -pe 's/ROOT\s/>dependo.1-whippomorpha-ancestral\n/' >dependo.1-whippomorpha.fasta

```

What does it look like?

```{r}
t1<-read.tree("outputs/201/RAxML_bipartitions.dependo.1") 
ggtree(t1) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,.4) 
```

### dependo.2
Eptesicus_fuscus_Dependo_120_,Lasiurus_borealis_Dependo_179_,Pipistrellus_pipistrellus_Dependo_295_    
Appear to not be monophyletic in some of the bootstrap trees. What about the "best" tree?    
```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Eptesicus_fuscus_Dependo_120_,Lasiurus_borealis_Dependo_179_,Pipistrellus_pipistrellus_Dependo_295_ -n dependo.2 -s dependo.2-vespertilionidae.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.2 -s dependo.2-vespertilionidae.aln.phy -n dependo.2.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.2.ancestral | perl -pe 's/ROOT\s/>dependo.2-vespertilionidae-ancestral\n/' >dependo.2-vespertilionidae.fasta

```

```{r}
t2<-read.tree("outputs/201/RAxML_nodeLabelledRootedTree.dependo.2.ancestral")
ggtree(t2)%<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,12)
```

### dependo.14, 16,17
```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Cercocebus_atys_Dependo_71_ -n dependo.14 -s dependo.14-cercopithecidae.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.14 -s dependo.14-cercopithecidae.aln.phy -n dependo.14.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.14.ancestral | perl -pe 's/ROOT\s/>dependo.14-cercopithecidae-ancestral\n/' >dependo.14-cercopithecidae.fasta

raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Vulpes_vulpes_Dependo_358_ -n dependo.16 -s dependo.16-canid.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.16 -s dependo.16-canid.aln.phy -n dependo.16.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.16.ancestral | perl -pe 's/ROOT\s/>dependo.16-canid-ancestral\n/' >dependo.16-canid.fasta

raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Panthera_tigris_altaica_Dependo_271_,Puma_concolor_Dependo_321_,Felis_catus_Dependo_148_  -n dependo.17 -s dependo.17-carnivore.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.17 -s dependo.17-carnivore.aln.phy -n dependo.17.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.17.ancestral | perl -pe 's/ROOT\s/>dependo.17-carnivore-ancestral\n/' >dependo.17-carnivore.fasta
```

```{r}
t14<-read.tree("outputs/201/RAxML_nodeLabelledRootedTree.dependo.14.ancestral")
ggtree(t14) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,6)
```

```{r}
t16<-read.tree("outputs/201/RAxML_nodeLabelledRootedTree.dependo.16.ancestral")
ggtree(t16) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,6)
```

```{r}
t17<-read.tree("outputs/201/RAxML_nodeLabelledRootedTree.dependo.17.ancestral")
ggtree(t17) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,6)
```
### dependo.22

```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Desmodus_rotundus_Dependo_101_  -n dependo.22 -s dependo.22-laurasiatheria.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.22 -s  dependo.22-laurasiatheria.aln.phy -n dependo.22.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.22.ancestral | perl -pe 's/ROOT\s/>dependo.22-laurasiatheria-ancestral\n/' > dependo.22-laurasiatheria-ancestral.fasta
```

Not all the tips match perfectly.
```{r}
t22<-read.tree("outputs/201/RAxML_nodeLabelledRootedTree.dependo.22.ancestral")
ggtree(t22) +geom_nodelab()+geom_tiplab() + xlim(0,30)

ggtree(t22) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,34)
```

### dependo.23

```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Vicugna_pacos_Dependo_356_   -n dependo.23 -s dependo.23-camelidae.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.23 -s  dependo.23-camelidae.aln.phy -n dependo.23.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.23.ancestral | perl -pe 's/ROOT\s/>dependo.23-camelid-ancestral\n/' > dependo.23-camelidae-ancestral.fasta
```

```{r}
t23<-read.tree("outputs/201/RAxML_nodeLabelledRootedTree.dependo.23.ancestral")
ggtree(t23) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,6)
```

### dependo.26
With more than one sequence in this one, I'm not sure how to root it. Let's take a look.

```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Daubentonia_madagascariensis_Dependo_95_ -n dependo.26 -s dependo.26-lemuridae.aln.phy
```

```{r}
t26<-read.tree("outputs/201/RAxML_bipartitions.dependo.26")
ggtree(t26) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,.7)
```

Propithecus are sister to each other. Maybe it wouldn't be so bad to root by both Dubentonia, though, it looks like there is some support for it to be with (Eulemur+Prolemur).

### dependo.30
This one is interesting. It looks like the EVE tripled in descendant taxa ?

```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Pteronotus_parnellii_Dependo_312_ -n dependo.30 -s dependo.30-chiroptera-bat1.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.30 -s  dependo.30-chiroptera-bat1.aln.phy -n dependo.30.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.30.ancestral | perl -pe 's/ROOT\s/>dependo.30-chiroptera-bat1-ancestral\n/' > dependo.30-chiroptera-bat1-ancestral.fasta
```

```{r}
t30<-read.tree("outputs/201/RAxML_bipartitions.dependo.30")
ggtree(t30) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE)+ xlim(0,1)
```

### dependo.33

Eidolon_helvum_Dependo_114
```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Eidolon_helvum_Dependo_114_ -n dependo.33 -s dependo.33-pteropodidae-bat4.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.33 -s  dependo.33-pteropodidae-bat4.aln.phy -n dependo.33.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.33.ancestral | perl -pe 's/ROOT\s/>dependo.34-pteropodidae-bat4-ancestral\n/' > dependo.33-pteropodidae-bat4-ancestral.fasta
```

```{r}
t33<-read.tree("outputs/201/RAxML_bipartitions.dependo.33")
ggtree(t33) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,0.5)
```

### dependo.34
Hmm... let's see if Megaderma_lyra can root the tree?

Yep!! Only occassionally non-monophyletic
```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Megaderma_lyra_Dependo_205_,Megaderma_lyra_Dependo_207_,Megaderma_lyra_Dependo_209_ -n dependo.34 -s dependo.34-megadermatvespertilion-bat5.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.34 -s  dependo.34-megadermatvespertilion-bat5.aln.phy -n dependo.34.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.34.ancestral | perl -pe 's/ROOT\s/>dependo.34-megadermatvespertilion-bat5-ancestral\n/' > dependo.34-megadermatvespertilion-bat5-ancestral.fasta
```

```{r}
t34<-read.tree("outputs/201/RAxML_bipartitions.dependo.34")
ggtree(t34) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,1.25)
```

### dependo.36
We ran root by one of the two major clades in bats it looks like.    
Eptesicus_fuscus_Dependo_119_,Desmodus_rotundus_Dependo_101_

Ahh, they don't seem to follow the taxonomy.
```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Eptesicus_fuscus_Dependo_119_,Desmodus_rotundus_Dependo_101_ -n dependo.36 -s dependo.36-vespertilionidae-batPCDH19.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.36 -s  dependo.36-vespertilionidae-batPCDH19.aln.phy -n dependo.36.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.36.ancestral | perl -pe 's/ROOT\s/>dependo.34-megadermatvespertilion-bat5-ancestral\n/' > dependo.36-vespertilionidae-batPCDH19-ancestral.fasta
```

```{r}
t36<-read.tree("outputs/201/RAxML_bipartitions.dependo.36")
ggtree(t36) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,.75)
```

### dependo.45

Cavia_porcellus_Dependo_64_    
```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Cavia_porcellus_Dependo_64_ -n dependo.45 -s dependo.45-rodent.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.45 -s  dependo.45-rodent.aln.phy -n dependo.45.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.45.ancestral | perl -pe 's/ROOT\s/>dependo.45-rodent-ancestral\n/' > dependo.45-rodent-ancestral.fasta
```

```{r}
t45<-read.tree("outputs/201/RAxML_bipartitions.dependo.45")
ggtree(t45) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,1.5)
```

## dependo.46

```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Apodemus_speciosus_Dependo_30_,Apodemus_sylvaticus_Dependo_31_  -n dependo.46 -s dependo.46-glires.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.46 -s dependo.46-glires.aln.phy -n dependo.46.ancestral
grep ROOT RAxML_marginalAncestralStates.dependo.46.ancestral | perl -pe 's/ROOT\s/>dependo.46-glires-ancestral\n/' > dependo.46-glires-ancestral.fasta

```

```{r}
t46<-read.tree("outputs/201/RAxML_bipartitions.dependo.46")
ggtree(t46) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,1)
```

### dependo.50

```{sh, eval=FALSE}

raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Peromyscus_maniculatus_bairdii_Dependo_283,Neotoma_lepida_Dependo_253 -n dependo.50 -s dependo.50-muroidae.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.50 -s dependo.50-muroidae.aln.phy -n dependo.50.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.50.ancestral | perl -pe 's/ROOT\s/>dependo.50-muroidae-ancestral\n/' > dependo.50-muroidae-ancestral.fasta
```

```{r}

t50<-read.tree("outputs/201/RAxML_bipartitions.dependo.50")
t50$tip.label<-paste0(t50$tip.label,"_")
ggtree(t50) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,.5)
```

### dependo.51

Mus_pahari_Dependo_237_

```{sh, eval=FALSE}

raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Mus_pahari_Dependo_237_ -n dependo.51 -s dependo.51-muridae.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.51 -s dependo.51-muridae.aln.phy -n dependo.51.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.51.ancestral | perl -pe 's/ROOT\s/>dependo.51-muridae-ancestral\n/' > dependo.51-muridae-ancestral.fasta
```

```{r}
t51<-read.tree("outputs/201//RAxML_bipartitions.dependo.51")
ggtree(t51) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,.15)
```

### dependo.53
Fukomys_damarensis_Dependo_150_

```{sh, eval=FALSE}

raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Fukomys_damarensis_Dependo_150_ -n dependo.53 -s dependo.53-cavidae.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.53 -s dependo.53-cavidae.aln.phy -n dependo.53.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.53.ancestral | perl -pe 's/ROOT\s/>dependo.53-cavidae-ancestral\n/' > dependo.53-cavidae-ancestral.fasta
```

```{r}
t53<-read.tree("outputs/201/RAxML_bipartitions.dependo.53")
ggtree(t53) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,.75)

ggsave("outputs/201/RAxML_bipartitions.dependo.53.jpeg")
```

### dependo.54

```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Dolichotis_patagonum_PVJX010010833 -n dependo.54 -s dependo.54-cavidae.aln.phy
raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.54 -s dependo.54-cavidae.aln.phy -n dependo.54.ancestral
grep ROOT RAxML_marginalAncestralStates.dependo.54.ancestral | perl -pe 's/ROOT\s/>dependo.54-cavidae-ancestral\n/' > dependo.54-cavidae-ancestral.fasta
```

The names in the alignments are really different. I'm not sure what to do about this one.

```{r}

df %>% filter(locus_name=="dependo.54-cavid") %>% select(id, scaffold, extract_start)

t54<-read.tree("outputs/201/RAxML_bipartitions.dependo.54")
ggtree(t54)+geom_nodelab()+geom_tiplab()+xlim(0,.5)
```

### dependo.65
Cicinnurus_regius_Dependo_84_,Diphyllodes_magnificus_Dependo_109_,Parotia_lawesii_Dependo_277_

```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Cicinnurus_regius_Dependo_84_,Diphyllodes_magnificus_Dependo_109_,Parotia_lawesii_Dependo_277_ -n dependo.65 -s dependo.65-passeriformes.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.65 -s dependo.65-passeriformes.aln.phy -n dependo.65.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.65.ancestral | perl -pe 's/ROOT\s/>dependo.65-passeriformes-ancestral\n/' > dependo.65-passeriformes-ancestral.fasta
```

```{r}
t65<-read.tree("outputs/201/RAxML_bipartitions.dependo.65")
ggtree(t65) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,.5)
```

### dependo.66

```{sh, eval=FALSE}
raxmlHPC-PTHREADS-SSE3 -T 3 -m GTRGAMMA -p 123 -f a -# 100 -x 123 -o Empidonax_traillii_Dependo_117_,Lepidothrix_coronata_Dependo_180_ -n dependo.66 -s dependo.66-passeriformes.aln.phy

raxmlHPC-PTHREADS-SSE3  -T 3 -m GTRGAMMA -p 123 -f A -t RAxML_bestTree.dependo.66 -s dependo.66-passeriformes.aln.phy -n dependo.66.ancestral

grep ROOT RAxML_marginalAncestralStates.dependo.66.ancestral | perl -pe 's/ROOT\s/>dependo.66-passeriformes-ancestral\n/' > dependo.66-passeriformes-ancestral.fasta
```

Like I said earlier, we're misssing one of the passeriforms from the time tree.
```{r}
t66<-read.tree("outputs/201/RAxML_bipartitions.dependo.66")
ggtree(t66) %<+% df + geom_nodelab() + geom_tiplab(aes(label=TipLabel), align=TRUE) + xlim(0,.5)
```